R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,407 @@
+                        linetype = "dashed") +
+    ggplot2::scale_y_continuous(breaks= jaspGraphs::getPrettyAxisBreaks(1:options$nReps)) +
+    ggplot2::xlab(gettextf("Observed mean and %s%% CI", options$confidenceIntervalInterval * 100)) 
+  
+  if (options$fixAxisTreePlot) {
+    p <- p + ggplot2::scale_x_continuous(limits = c(options$fixAxisLower, options$fixAxisUpper))
+  }
+  
+  p <- p +
+    jaspGraphs::geom_rangeframe() + # add lines on the x-axis and y-axis
+    jaspGraphs::themeJaspRaw()      # add the JASP theme
+  
+  if(isTryError(p))
+    treePlotContainer[["treePlot"]]$setError(.extractErrorMessage(p))
+  else
+    treePlotContainer[["treePlot"]]$plotObject <- p
+  
+  return()
+}
+
+.plotDatasetsOLD <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["containerRainCloudPlots"]]) || !options$dataPlot)
+    return()
+  
+  rainCloudPlotsContainer <- createJaspContainer(title = gettext("Data plots"))
+  # rainCloudPlotsContainer$position <- 3
+  jaspContainer[["containerRainCloudPlots"]] <- rainCloudPlotsContainer
+  rainCloudPlotsContainer$dependOn(c("dataPlotShowN", "dataPlot"))
+  
+  rainData <- data.frame(y = unlist(jaspContainer[["simulatedDatasets"]][["object"]]), 
+                         group = rep(1:options$nReps, each = options$n))
+  
+  singlePlot <- createJaspPlot(title = "", width = 480, height = 320)
+  rainCloudPlotsContainer[["rainCloudPlotAll"]] <- singlePlot
+  p <- try(jaspTTests::.descriptivesPlotsRainCloudFill(rainData, "y", "group", 
+                                                       yLabel = "Dependent", 
+                                                       xLabel = "Rep", 
+                                                       testValue = options$mu,
+                                                       addLines = FALSE, horiz = FALSE))
+  if(isTryError(p))
+    singlePlot$setError(.extractErrorMessage(p))
+  else
+    singlePlot$plotObject <- p
+  
+  if (options$dataPlotShowN > 1) {
+    for(i in 1:options$dataPlotShowN) {
+      
+      thisRainData <- data.frame(y = unlist(jaspContainer[["simulatedDatasets"]][["object"]][[i]]), 
+                                 group = rep(1, options$n))
+      
+      singlePlot <- createJaspPlot(title = "", width = 480, height = 320)
+      rainCloudPlotsContainer[[paste0("rainCloudPlotSingle", i)]] <- singlePlot
+      p <- try(jaspTTests::.descriptivesPlotsRainCloudFill(thisRainData, "y", "group", 
+                                                           yLabel = "Dependent", 
+                                                           xLabel = "Rep",
+                                                           testValue = options$mu,
+                                                           addLines = FALSE, horiz = FALSE))
+      if(isTryError(p))
+        singlePlot$setError(.extractErrorMessage(p))
+      else
+        singlePlot$plotObject <- p
+      
+    }
+    
+  }
```

Comment:
```suggestion
  for (i in seq_len(options$dataPlotShowN)) {

      
    thisRainData <- data.frame(
      y = unlist(jaspContainer[["simulatedDatasets"]][["object"]][[I]]), 
      group = rep(1, options$n)
    )
      
    singlePlot <- createJaspPlot(title = "", width = 480, height = 320)
    rainCloudPlotsContainer[[paste0("rainCloudPlotSingle", i)]] <- singlePlot
    p <- try(jaspTTests::.descriptivesPlotsRainCloudFill(thisRainData, "y", "group", 
                                                         yLabel = "Dependent", 
                                                         xLabel = "Rep",
                                                         testValue = options$mu,
                                                         addLines = FALSE, horiz = FALSE))
    if(isTryError(p))
      singlePlot$setError(.extractErrorMessage(p))
    else
      singlePlot$plotObject <- p
      
  }
```
this is one of those cases where `seq_len` automatically does the right thing. The loop will not run if `options$dataPlotShowN < 1`.